dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -53,15 +59,68 @@ class TypedDataset[T](
     }
   }
 
-  /**
-    * Job returns an array that contains all the elements in this [[TypedDataset]].
+  /** Job returns an array that contains all the elements in this [[TypedDataset]].
     *
     * Running job requires moving all the data into the application's driver process, and
     * doing so on a very large [[TypedDataset]] can crash the driver process with OutOfMemoryError.
+    *
+    * Differs from `Dataset#collect` by wrapping it's result into a [[frameless.Job]].
+    */
+  def collect(): Job[Array[T]] =
```

Comment:
AFAIK Array doesn't extend `Seq`, there is implicit to convert it to `WrappedArray`, which is mutable too. I don't think it makes any difference for collections under 1M elements, I would prefer safety over performance in this case.